src/coreclr/jit/hwintrinsiclistarm64sve.h

```diff
@@ -109,6 +109,16 @@ HARDWARE_INTRINSIC(Sve,           MultiplySubtract,
 HARDWARE_INTRINSIC(Sve,           Negate,                                                           -1,      -1,     false, {INS_sve_neg,        INS_invalid,        INS_sve_neg,        INS_invalid,        INS_sve_neg,        INS_invalid,        INS_sve_neg,        INS_invalid,        INS_sve_fneg,       INS_sve_fneg},    HW_Category_SIMD,                  HW_Flag_Scalable|HW_Flag_EmbeddedMaskedOperation|HW_Flag_LowMaskedOperation)
 HARDWARE_INTRINSIC(Sve,           Or,                                                               -1,      -1,     false, {INS_sve_orr,        INS_sve_orr,        INS_sve_orr,        INS_sve_orr,        INS_sve_orr,        INS_sve_orr,        INS_sve_orr,        INS_sve_orr,        INS_invalid,        INS_invalid},     HW_Category_SIMD,                  HW_Flag_Scalable|HW_Flag_OptionalEmbeddedMaskedOperation|HW_Flag_HasRMWSemantics|HW_Flag_LowMaskedOperation)
 HARDWARE_INTRINSIC(Sve,           OrAcross,                                                         -1,      -1,     false, {INS_sve_orv,        INS_sve_orv,        INS_sve_orv,        INS_sve_orv,        INS_sve_orv,        INS_sve_orv,        INS_sve_orv,        INS_sve_orv,        INS_invalid,        INS_invalid},     HW_Category_SIMD,                  HW_Flag_Scalable|HW_Flag_EmbeddedMaskedOperation|HW_Flag_LowMaskedOperation)
+
+HARDWARE_INTRINSIC(Sve,           SaturatingDecrementBy16BitElementCount,                           -1,      3,      true,  {INS_invalid,        INS_invalid,        INS_sve_sqdech,     INS_sve_uqdech,     INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid},     HW_Category_SIMD,                  HW_Flag_Scalable|HW_Flag_HasImmediateOperand|HW_Flag_HasEnumOperand|HW_Flag_SpecialCodeGen|HW_Flag_HasScalarVariant|HW_Flag_NoJmpTableIMM|HW_Flag_HasRMWSemantics)
```

Comment:
`HWIntrinsicInfo::NoJmpTableImm()` is used in a few places. It ensures the intrinsic doesn't generate a fallback table if the inputs aren't constants. Eg in `CheckHWIntrinsicImmRange()`

Reply:
right, but I don't see the usage of `NoJmpTableImm()` being used for `Arm64`? am I missing something?

Reply:
In `Compiler::CheckHWIntrinsicImmRange()`:

```
else if (!immOp->IsCnsIntOrI())
    {
        if (HWIntrinsicInfo::NoJmpTableImm(intrinsic))
        {
            *useFallback = true;
            return false;
        }
```
That's the clause that will cause it to fallback to the C# implementation.

Reply:
> It ensures the intrinsic doesn't generate a fallback table if the inputs aren't constants.

The point of this is when there is an alternative instruction that allows for identical behavior. For example, on x64 `ShiftLeftLogical` has two forms one that allows the amount to be encoded as an n-bit immediate value and another that takes the amount via a SIMD register. Thus, instead of generating a fallback table in this case we can emit the alternative instruction encoding with identical behavior instead.

It is not to allow a pseudo-implementation to be provided in C#.